### ISSUE TITLE
[Optimization] R0_and_mean_shape_estimation.R - Special-case zero length data

### ISSUE THREAD

**Issue by Siddhant** [sourish-cmi/Covid19#3]
https://github.com/sourish-cmi/Covid19/blob/a4fe6010b18872d65b51c1c17ba99914dba61240/R0_and_mean_shape_estimation.R#L56 

We can skip this loop for States/UT who (thankfully) have no cases, or date range having no cases. No need of unnecessarily doing N^2 iterations.

Consider directly returning 'NA' values for R0, maybe?

**Comment by sourish-cmi** [comment]
Yes ... N^2 is a big headache... for some choices of mean and shape it is
returning NA. Thinking of binary search but not sure

**Comment by sourish-cmi** [comment]
Agree ... a check will work
